### PR TITLE
Add gzip capability to HTTP endpoints

### PR DESCRIPTION
**Pull request by corylanou**
Fixes https://github.com/influxdb/influxdb/issues/1449

### CODE REVIEW

**Comment by otoolep** [comment]
Makes sense, but is there no way to unit-test this? Surely there is.

File: httpd/handler.go [hunk 1]
```diff
@@ -376,6 +380,29 @@ func authenticate(inner func(http.ResponseWriter, *http.Request, *influxdb.User)
...
+
+func (w gzipResponseWriter) Write(b []byte) (int, error) {
+	return w.Writer.Write(b)
+}
+
+// determines if the client can accept compressed responses, and encodes accordingly
+func gzipFilter(inner http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+			inner.ServeHTTP(w, r)
```
Comment (otoolep) [inline httpd/handler.go]: To be clear about this, `inner.ServerHTTP` will be called if `gzip` is not supported, then the `else` will be supported. Should it instead return?

Comment (corylanou) [inline httpd/handler.go]: Return added.

**Comment by corylanou** [comment]
Unit tests added.

File: httpd/handler_test.go [hunk 2]
```diff
@@ -480,6 +480,54 @@ func TestHandler_DeleteRetentionPolicy_NotFound(t *testing.T) {
...
+	s := NewHTTPServer(srvr)
+	defer s.Close()
+
+	req, err := http.NewRequest("GET", s.URL+`/ping`, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept-Encoding", "")
```
Comment (otoolep) [inline httpd/handler_test.go]: Not sure you even need to do this. No guarantee a client will include this header AFAIK.

**Comment by otoolep** [comment]
Looks great, +1, assuming all is good even without the header. Thanks for the tests.

File: httpd/handler_test.go (hunk 2 above)
Comment (corylanou) [inline httpd/handler_test.go]: We are actually working against the HTTP library in Go.  It supports gzip automatically, so if I don't "unset" it in the test, it's always set to `gzip`.  This mimics the real world for when the don't send the header.

Comment (otoolep) [inline httpd/handler_test.go]: Cool, makes sense.

**Pull request merged**